src/mono/wasm/Wasm.Build.Tests/TestAppScenarios/BlazorHostedApp/SignalRClientTests.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Playwright;
+using Xunit.Abstractions;
+using Xunit;
+
+#nullable enable
+
+namespace Wasm.Build.Tests.TestAppScenarios;
+
+public class SignalRClientTests : AppTestBase
+{
+    public SignalRClientTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+        : base(output, buildContext)
+    {
+    }
+
+    // ToDo: run only on MT
+    [Fact]
+    public async Task SignalRClientWorksWithLongPolling()
+    {
+        CopyTestAsset("BlazorHostedApp", "SignalRClientTests");
+
+        string config = "Debug";
+        RunOptions options = new(
+            Configuration: config,
+            // for now, TestScenario is ignored - there's only one scenario
+            TestScenario: "SignalRClientTests",
+            BrowserQueryString: new Dictionary<string, string> { ["message"] = "test",  ["transport"] = "LongPolling" },
+            ExtraArgs: $"--logRootPath {s_buildEnv.LogRootPath}"
+        );
+
+        string rootProjectPath = Directory.GetParent(_projectDir!)?.FullName ?? "";
+        string clientProjectDir = Path.Combine(rootProjectPath, "BlazorHosted.Client");
+        string frameworkDir = FindBlazorBinFrameworkDir(config, forPublish: false, projectDir: clientProjectDir);
```

Comment:
Why do we need this information? Building/running from the hosted project dir should be enough, shouldn't it?

Reply:
This way we assert that on checking `AppBundle` we will find `_framework`. Without this, it's looking for `_framework` folder in `BlazorHosted.Server` because we set the `_projectDir` as `BlazorHosted.Server` in `CopyTestAsset`. We have to set `_projectDir` to `BlazorHosted.Server` because that's where the starting project is.

Reply:
I would prefer to hide these details from individual tests. It's specific to the test asset, not the actual test, So my ideas
a) Compute this path in the `CopyTestAsset` similar to how we modify the `_projectDir` based on individual asset
b) Instead of having a string for testasset, have some class with metadata about the test asset specifics and pass some static instance of such class to `CopyTestAsset`